### PR TITLE
all project

### PR DESCRIPTION
**Pull request by maksimk99**
It is all what I already have.
Then I will try to break other pull requests into parts.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  <artifactId>ListOfGoods</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>ListOfGoods</name>
+  <!-- FIXME change it to the project's website -->
+  <url>http://www.example.com</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (aleksieiev) [inline pom.xml]: why 1.7?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.spring.core</groupId>
+  <artifactId>ListOfGoods</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>ListOfGoods</name>
+  <!-- FIXME change it to the project's website -->
```
Comment (aleksieiev) [inline pom.xml]: wtf and what for?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  <url>http://www.example.com</url>
```
Comment (aleksieiev) [inline pom.xml]: are you sure that it's correct URL of your project?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.7</maven.compiler.source>
+    <maven.compiler.target>1.7</maven.compiler.target>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.12</version>
```
Comment (aleksieiev) [inline pom.xml]: version should be provided as a constant from properties.

File: src/main/java/com/spring/core/config/DataInitializer.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This class works with json file and gets data from it
+ *
+ * @author Katuranau Maksimilyan
+ */
+@Component
+public class DataInitializer {
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/config/DataInitializer.java]: Don't think that it's required functionality according to the task.
Show me, please, where it is in the task?

File: src/main/java/com/spring/core/dao/BasketDao.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.List;
+
+/**
+ * this interface provides methods to manage products basket
+ * @author Katuranau Maksimilyan
+ */
+public interface BasketDao {
+    /**
+     *
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/dao/BasketDao.java]: where is method details?

File: src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class implements BasketDao interface which provides methods to manage products basket
+ * @see BasketDao <-- There
+ * @author Katuranau Maksimilyan
+ */
+public class InMemoryBasketDaoImpl implements BasketDao {
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java]: `InMemoryBasketDao` is better

File: src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.spring.core.dao.BasketDao;
+import com.spring.core.model.Product;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class implements BasketDao interface which provides methods to manage products basket
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java]: Here can be provided details about specific realization of this class, I mean that it's a InMemory storage

File: src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+ * @see BasketDao <-- There
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java]: `{@see BasketDao}` is more preferable

File: src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        basket.add(product);
+    }
+
+    @Override
+    public void deleteProduct(int index) {
+        basket.remove(index);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/dao/impl/InMemoryBasketDaoImpl.java]: Are you sure that you need these methods here?

File: src/main/java/com/spring/core/model/Product.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+     *
+     * @return String value
+     */
+    @Override
+    public String toString() {
+        return name + ": " + price + "$, " + weight + "g.";
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/model/Product.java]: are you sure that you need it?

File: src/main/java/com/spring/core/service/ProductService.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
+package com.spring.core.service;
+
+import com.spring.core.model.Product;
+
+import java.util.List;
+
+/**
+ *
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/service/ProductService.java]: JavaDoc

File: src/main/java/com/spring/core/userinterface/UserInterface.java [hunk 13]
```diff
@@ -0,0 +1,59 @@
+package com.spring.core.userinterface;
+
+import com.spring.core.userinterface.impl.UserMenuChoice;
+
+/**
+ * this interface provides methods to manage products basket from console
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/userinterface/UserInterface.java]: Interface shouldn't be depends on details like "from console"
It's an abstraction

File: src/test/java/com/spring/core/BasketTest.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+import com.spring.core.model.Product;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class BasketTest {
```
Comment (aleksieiev) [inline src/test/java/com/spring/core/BasketTest.java]: Only one negative test.
It's ugly

Comment (maksimk99) [reply src/test/java/com/spring/core/BasketTest.java]: Check last commit: 
1) I changed project structure:
now there is object with list of goods which creates when project start an you can add product from it to your basket
2) change some code and javaDoc that you said

*I'm adding more tests and read that you said

**Comment by maksimk99** [comment]
I made integration test and changed type of price to BigDecimal
I have nothing more to do
I can read now about 'loger'

File: src/test/java/com/spring/core/UstUserInterfaceTest.java [hunk 15]
```diff
@@ -21,7 +21,7 @@
 import static org.mockito.Mockito.*;
 
 @RunWith(MockitoJUnitRunner.class)
-public class UserInterfaceTest {
+public class UstUserInterfaceTest {
 
     private ApplicationContext context;
```
Comment (aleksieiev) [inline src/test/java/com/spring/core/UstUserInterfaceTest.java]: what for is it here?

File: src/test/java/com/spring/core/UstUserInterfaceTest.java [hunk 16]
```diff
@@ -21,7 +21,7 @@
 import static org.mockito.Mockito.*;
 
 @RunWith(MockitoJUnitRunner.class)
-public class UserInterfaceTest {
+public class UstUserInterfaceTest {
```
Comment (aleksieiev) [inline src/test/java/com/spring/core/UstUserInterfaceTest.java]: It's a unit test.
You do not need to have here ApplicationContext

Comment (aleksieiev) [reply src/test/java/com/spring/core/UstUserInterfaceTest.java]: basketService should be mock

File: ListOfGoods.iml [hunk 17]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aleksieiev) [inline ListOfGoods.iml]: IML file shouldn't been committed at all. It's Idea file

File: src/main/java/com/spring/core/console_input/ConsoleInputValidator.java [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.InputStream;
+import java.util.Scanner;
+
+/**
+ * This class is used to enter information from the stream
+ *
+ * @author Katuranau Maksimilyan
+ */
+
+public class ConsoleInputValidator {
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/console_input/ConsoleInputValidator.java]: Maybe it should be a bean?

File: src/main/java/com/spring/core/service/BasketService.java [hunk 19]
```diff
@@ -0,0 +1,50 @@
+package com.spring.core.service;
+
+import com.spring.core.model.Product;
+
+import java.util.List;
+
+/**
+ * Use this interface if you want to create a new service
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/service/BasketService.java]: wtf?
what kind of a service? What is this service for?

File: src/main/java/com/spring/core/service/impl/BasketServiceImpl.java [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+import com.spring.core.dao.BasketDao;
+import com.spring.core.model.Product;
+import com.spring.core.service.BasketService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * This class provides all the functionality
```
Comment (aleksieiev) [inline src/main/java/com/spring/core/service/impl/BasketServiceImpl.java]: All functionality?
Can it make a coffee for me?

File: src/test/java/com/spring/core/BasketServiceTest.java [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+    @BeforeClass
+    public static void initialProductService() throws Exception {
+        list = new ArrayList<>() {{
+            add(new Product("phone", new BigDecimal(790.22), 340));
+            add(new Product("pan", new BigDecimal(9.12), 15));
+            add(new Product("computer", new BigDecimal(3200.00), 2100));
+        }};
+
+        listOfGoods = Mockito.mock(ListOfGoodsDao.class);
+        basket = new InMemoryBasketDao(listOfGoods);
```
Comment (aleksieiev) [inline src/test/java/com/spring/core/BasketServiceTest.java]: Think it should be Mock

File: src/test/java/com/spring/core/BasketServiceTest.java [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+        when(listOfGoods.getProduct(1)).thenReturn(list.get(1));
+        when(listOfGoods.getProduct(2)).thenReturn(list.get(2));
+        for (int i = 0; i < 3; i++) {
+            basketService.addProduct(i);
+        }
+    }
+
+    @Test
+    public void testGetProductsFromBasket() {
+        assertEquals(list, basketService.getProductsFromBasket());
```
Comment (aleksieiev) [inline src/test/java/com/spring/core/BasketServiceTest.java]: how it works? Do these lists really equals or just links are equals?
think it's  incorrect validation

File: src/test/java/com/spring/core/UserInterfaceTestIT.java [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+    @Test
+    public void testPrintProductsFromBasket() {
+        try {
+            when(inputStream.nextInt()).thenReturn(3, 7);
+            userInterface.addProduct();
+            userInterface.addProduct();
+            int size = basketService.getProductsFromBasket().size();
+            userInterface.printProductsFromBasket();
+            verify(outputStream, times(size)).println(anyString());
+        } catch (Exception e) {
```
Comment (aleksieiev) [inline src/test/java/com/spring/core/UserInterfaceTestIT.java]: It's a positive test case. What kind of exceptions are you expecting here?

**Comment by aleksieiev** [comment]
@maksimk99 please merge and update your next PR if it's required.

**Comment by aleksieiev** [comment]
@maksimk99 
Merge it and update your next PR, ping me after that for review

**Pull request merged**